pepys_import/core/store/data_store.py

```diff
@@ -1215,33 +1223,160 @@ def get_cached_platform_name(self, sensor_id=None, platform_id=None):
+                assert len(result) == 3, (
+                    "It should return minimum date, maximum date and datafile" " id in a row!"
+                )
+                min_time, max_time, datafile_id = result
+                # Extract datafile name from the given datafile id
+                datafile_name = self.get_datafile_from_id(datafile_id).reference
+
+            # Find minimum date, maximum date, and datafile name of the filtered Contact objects
+            result = self.find_min_and_max_date(Contact, Contact.sensor_id, sensor_id)
+            min_time_2, max_time_2 = datetime.utcnow(), datetime(day=1, month=1, year=1700)
+            if result:
+                assert len(result) == 3, (
+                    "It should return minimum date, maximum date and datafile" " id in a row!"
+                )
+                min_time_2, max_time_2, datafile_id_2 = result
+                if not datafile_name:
+                    datafile_name = self.get_datafile_from_id(datafile_id_2).reference
+            # Compare min and max dates of State and Contact objects
+            min_, max_ = min(min_time, min_time_2), max(max_time, max_time_2)
+            # Append to list if every value is assigned
+            if sensor_name and datafile_name and min_ and max_:
+                objects.append(
+                    {
+                        "name": sensor_name,
+                        "filename": datafile_name,
+                        "min": str(min_),
+                        "max": str(max_),
+                        "sensor_id": sensor_id,
+                        "datafile_id": datafile_id or datafile_id_2,
+                    }
+                )
+        # Find minimum date, maximum date, and datafile name of the filtered Comment objects
+        comment_objects = self.find_min_and_max_date(Comment, Comment.platform_id, platform_id)
+        if comment_objects:
+            min_time, max_time, datafile_id = comment_objects
+            datafile_name = self.get_datafile_from_id(datafile_id).reference
+            objects.append(
+                {
+                    "name": "Comment",
+                    "filename": datafile_name,
+                    "min": min_time,
+                    "max": max_time,
+                    "platform_id": platform_id,
+                    "datafile_id": datafile_id,
+                }
+            )
+
+        return objects
+
+    def export_datafile(self, datafile_id, file_path, sensor_id=None, platform_id=None):
         """Gets states, contacts and comments of a Datafile.
 
         :param datafile_id:  ID of Datafile
         :type datafile_id: Integer or UUID
         :param file_path: Path of a file to export
         :type file_path: String
+        :param sensor_id: ID of Sensor to export a specific sensor in the datafile, default is None
+        :type sensor_id: Integer or UUID
+        :param platform_id: ID of Sensor to export a specific sensor in the datafile, default is None
```

Comment:
I think the parameter description is wrong, I presume it's the platform.

Reply:
Yes, it was a copy-paste error. Corrected:
https://github.com/debrief/pepys-import/pull/266/commits/64872c9671e6f6511f50d3b66dae0e745057df05